warmrestart/warmRestartHelper.cpp

```diff
@@ -0,0 +1,352 @@
+
+            m_syncTable->del(restoredKey);
+        }
+
+        /*
+         * If a matching entry is found in refreshMap, proceed to compare it
+         * with its restored counterpart.
+         */
+        else
+        {
+            auto refreshedKey = kfvKey(iter->second);
+            auto refreshedFV  = kfvFieldsValues(iter->second);
+
+            if (compareAllFV(restoredFV, refreshedFV))
+            {
+                SWSS_LOG_NOTICE("Warm-Restart reconciliation: updating entry %s",
+                                printKFV(refreshedKey, refreshedFV).c_str());
+
+                m_syncTable->set(refreshedKey, refreshedFV);
+            }
+            else
+            {
+                SWSS_LOG_INFO("Warm-Restart reconciliation: no changes needed for "
+                              "existing entry %s",
+                              printKFV(refreshedKey, refreshedFV).c_str());
+            }
+        }
+
+        /* Deleting the just-processed restored entry from the refreshMap */
+        m_refreshMap.erase(restoredKey);
+    }
+
+    /*
+     * Iterate through all the entries left in the refreshMap, which correspond
+     * to brand-new entries to be pushed down to AppDB.
+     */
+    for (auto &kfv : m_refreshMap)
+    {
+        auto refreshedKey = kfvKey(kfv.second);
+        auto refreshedFV  = kfvFieldsValues(kfv.second);
+
+        SWSS_LOG_NOTICE("Warm-Restart reconciliation: introducing new entry %s",
+                        printKFV(refreshedKey, refreshedFV).c_str());
+
+        m_syncTable->set(refreshedKey, refreshedFV);
+    }
+
+    /* Clearing pending kfv's from refreshMap */
+    m_refreshMap.clear();
+
+    /* Clearing restoration vector */
+    m_restorationVector.clear();
+
+    setState(WarmStart::RECONCILED);
+
+    SWSS_LOG_NOTICE("Warm-Restart: Concluded reconciliation process for %s "
+                    "application.", m_appName.c_str());
+}
+
+
+/*
+ * Compare all field-value-tuples within two vectors.
+ *
+ * Example: v1 {nexthop: 10.1.1.1, ifname: eth1}
+ *          v2 {nexthop: 10.1.1.2, ifname: eth2}
+ *
+ * Returns:
+ *
+ *    'false' : If the content of both 'fields' and 'values' fully match
+ *    'true'  : No full-match is found
+ */
+bool WarmStartHelper::compareAllFV(const std::vector<FieldValueTuple> &v1,
+                                   const std::vector<FieldValueTuple> &v2)
+{
+    std::unordered_map<std::string, std::string> v1Map((v1.begin()), v1.end());
+
+     /* Iterate though all v2 tuples to check if their content match v1 ones */
+    for (auto &v2fv : v2)
+    {
+        auto v1Iter = v1Map.find(v2fv.first);
+        /*
+         * The sizes of both tuple-vectors should always match within any
+         * given application. In other words, all fields within v1 should be
+         * also present within v2. Otherwise we are running into some form of
+         * bug or an unsupported functionality.
+         */
+        assert(v1Iter != v1Map.end());
```

Comment:
What if v1 is superset of v2?

Reply:
If by 'superset' you're referring to v1 having more fields than v2, that shouldn't ever happen as you commented further below. I'll add more details to my code-comments about this expectation.